File: src/test/java/dev/kensa/SourceCodeIndexTest.java
package dev.kensa;

import dev.kensa.util.SourceCodeIndex;
import org.junit.jupiter.api.Test;

import java.nio.file.Path;

import static org.assertj.core.api.Assertions.assertThat;

class SourceCodeIndexTest {


    @Test
    void canLocateByInnerClass() {
        Path path = SourceCodeIndex.locate(InnerClass.class);

        assertThat(path).isNotNull();
    }


    static class InnerClass {

    }
}

File: src/test/java/dev/kensa/SourceCodeIndexTest.kt
package dev.kensa

import dev.kensa.util.SourceCodeIndex.locate
import io.kotest.matchers.string.shouldEndWith
import org.junit.jupiter.api.Test
import kotlin.io.path.absolutePathString

internal class SourceCodeIndexTest {
    @Test
    fun canLocateByInnerClass() {
        locate(InnerClass::class.java).absolutePathString() shouldEndWith "/src/test/java/dev/kensa/SourceCodeIndexTest.kt"
    }

    internal class InnerClass
}